src/lib/png.ts

```diff
@@ -35,26 +35,45 @@ export const filterImagesBySizes = (images: ImageInfo[], sizes: number[]) => {
  * @param size The size (width/height) of the image.
  * @param dir Path of the file output directory.
  * @param logger Logger.
+ * @param autoMargin If true, than automargin the image.
  * @return Image generation task.
  */
 const generate = async (
   svg: Buffer,
   size: number,
   dir: string,
-  logger: Logger
+  logger: Logger,
+  autoMargin = false,
 ): Promise<ImageInfo> => {
   const dest = path.join(dir, size + '.png')
   logger.log('  Create: ' + dest)
 
+  const background = { r: 0, g: 0, b: 0, alpha: 0 };
+  let margin = 0;
+  let width = size;
+  let height = size;
+  if (autoMargin) {
+    margin = Math.max(0, (size - 16) / 8 + 1);
+    width = size - (margin * 2);
+    height = size - (margin * 2);
```

Comment:
It seems that let can be avoided by making it a function.
```suggestion
  const calcSizeWithMargin = (size: number, autoMargin: boolean)  => {
    if (!autoMargin) {
      return {
        width: size,
        height: size,
      }
    }

  const margin = Math.max(0, (size - 16) / 8 + 1)
    return {
      margin,
      width: size - (margin * 2),
      height: size - (margin * 2),
    }
  }

  const { width, height } = calcSizeWithMargin(size, autoMargin)
```

Reply:
`16`, `8`, etc. are magic numbers and should be named as constants.